Helm v3 docu (#8749)
kyma-project/kyma: docs/kyma/03-03-charts.md

## Notes

All notes are based on Helm v3.2.1 implementation and are subject to change in future releases.

* Regardless of how complex a chart is, and regardless of the number of sub-charts it references or consists of, it's always evaluated as one. This means that each Helm release is compiled into a single Kubernetes manifest file when applied on API server.

* Hooks are parsed in the same order as manifest files and returned as a single, global list for the entire chart. For each hook the weight is calculated as a part of this sort.

* Manifests are sorted by `Kind`. You can find the list and the order of the resources on the Helm [Github](https://github.com/helm/helm/blob/release-3.2/pkg/releaseutil/kind_sorter.go) page.

* To provide better error handling, Helm validates rendered templates against the Kubernetes OpenAPI schema before they are sent to the Kubernetes API. This means any resources that are not aligned with the Kubernetes API docs (e.g. because of unsupported fields) will fail the release.

## Migration to Helm v3

> Beginning with the version 1.14, Kyma uses [Helm v3](https://helm.sh/) to install and maintain components. To ensure compatibility, any Helm v2 releases existing on your cluster are automatically migrated to v3 prior to the upgrade process. The migration is handled by the Helm [2to3](https://github.com/helm/helm-2to3) plugin that runs as an init container inside the Kyma Installer Pod. This means that the component may take more time than usual to start operating. If the plugin fails to migrate a release, the container exits with an error and the upgrade process is interrupted. Note that successful migration does not erase Helm v2 configuration and release data. Thus, you can still access pre-upgrade Helm resources using the v2 CLI if you establish a secure connection with Tiller. Read the [additional configuration](#installation-use-helm) document to learn how to access legacy resources using Helm v2 CLI.

Reviewer: ```suggestion
As of version 1.14, Kyma uses [Helm v3](https://helm.sh/) to install and maintain components. To ensure compatibility, any Helm v2 releases existing on your cluster are automatically migrated to v3 prior to the upgrade process. The migration is handled by the Helm [2to3](https://github.com/helm/helm-2to3) plugin that runs as an init container inside the Kyma Installer Pod. This means that it may take some more time for the component to start operating. If the plugin fails to migrate a release, the container exits with an error and the upgrade process is interrupted. 

>**NOTE:** Note that successful migration does not erase Helm v2 configuration and release data. Thus, you can still access pre-upgrade Helm resources using the v2 CLI if you establish a secure connection with Tiller. Read the [additional configuration](#installation-use-helm) document to learn how to access legacy resources using Helm v2 CLI.
```

* Manifests are sorted by `Kind`. You can find the list and the order of the resources on the Helm [Github](https://github.com/helm/helm/blob/release-3.2/pkg/releaseutil/kind_sorter.go) page.

* To provide better error handling, Helm validates rendered templates against the Kubernetes OpenAPI schema before they are sent to the Kubernetes API. This means any resources that don't comply with the Kubernetes API docs (for example because of unsupported fields) will fail the release.

## Migration to Helm v3

As of version 1.14, Kyma uses [Helm v3](https://helm.sh/) to install and maintain components. To ensure compatibility, any Helm v2 releases existing on your cluster are automatically migrated to v3 prior to the upgrade process. The migration is handled by the Helm [2to3](https://github.com/helm/helm-2to3) plugin that runs as an init container inside the Kyma Installer Pod. This means that it may take some more time for the component to start operating. If the plugin fails to migrate a release, the container exits with an error and the upgrade process is interrupted.

> You can skip the migration of custom components by extending the `CHART_BLACKLIST` variable in the [script](/installation/resources/installer.yaml) run by the init-container. If you decide to skip the migration of a given component, make sure to remove this component from the `Installation` custom resource.

Reviewer: ```suggestion
You can skip the migration of custom components by extending the **CHART_BLACKLIST** variable in the [script](/installation/resources/installer.yaml) run by the init-container. If you decide to skip the migration of a given component, make sure to remove this component from the `Installation` custom resource.
```

> The migration of Helm v2 components is a one-time event. Thus, the init-container will be removed in the following release.

Reviewer: ```suggestion
The migration of Helm v2 components is a one-time event. The init-container will be removed with the next release.
```

> >**NOTE:** Note that successful migration does not erase Helm v2 configuration and release data. Thus, you can still access pre-upgrade Helm resources using the v2 CLI if you establish a secure connection with Tiller. Read the [additional configuration](#installation-use-helm) document to learn how to access legacy resources using Helm v2 CLI. Alternatively, you can manually remove Tiller and Helm v2 release data after the migration using Helm's [2to3](https://github.com/helm/helm-2to3) plugin.

Reviewer: ```suggestion
>**NOTE:** Successful migration does not erase Helm v2 configuration and release data. You can still access pre-upgrade Helm resources using the v2 CLI if you establish a secure connection with Tiller. Read the [additional configuration](#installation-use-helm) document to learn how to access legacy resources using Helm v2 CLI. Alternatively, you can manually remove Tiller and Helm v2 release data after the migration using Helm's [2to3](https://github.com/helm/helm-2to3) plugin.
```
